app/helpers/elasticsearch_helper.rb

```diff
@@ -32,11 +33,32 @@ def taxon_search(prefix, tax_levels = TaxonCount::NAME_2_LEVEL.keys)
         }
       end
     end
-    matching_taxa.values
+
+    if project_id.present?
+      filter_by_project(matching_taxa.values, project_id)
+    else
+      matching_taxa.values
+    end
   end
 
   private
 
+  # Took 250ms in local testing on real data
+  def filter_by_project(matching_taxa, project_id)
+    # TODO: (gdingle): do we want both species and genus? only genus appears to be returned by TaxonCount
+    project_tax_ids = Set.new(TaxonCount
+      .joins(pipeline_run: { sample: :project })
+      .where(samples: { project_id: project_id })
+      .where(tax_id: [matching_taxa.map { |taxa| taxa["taxid"] }])
+      .where("tax_id > 0") # negative numbers mean unknown... see TaxonLineage
```

Comment:
should we also apply this to elasticsearch above?

Reply:
I think it's unnecessary... if searches by name, so if it has a name for a level than it also has a positive taxid